src/main/java/core/Game.java

```diff
@@ -495,13 +502,93 @@ public final void run() {
         }
     }
 
+    /**
+     * Runs an instance of the game, with the possibility to run fully parallel from any other game instances.
+     * Not to be used when there is a human player, only useful when parallelization needs to be possible.
+     * @return The final gameState object after finishing the game run(s)
+     */
+    public AbstractGameState runInstance(LinkedList<AbstractPlayer> players, int seed, boolean randomGameParameters) {
```

Comment:
This is essentially the same method as the `run()` method, except that it facilitates parallel execution. The reason this is a separate method, and not integrated into `run()` is because this function requires as few `synchronized` parts, in order to speed up execution, whereas `run()` requires a large synchronized block of code to facilitate GUI interaction such as pausing.
This meant I unfortunately needed to duplicate some of the code, although I removed all of the unnecessary pause/stop parts of the code. However, this also meant that most of the multithreading-specific code (such as increased number of scoped variables, instead of using class variables) is also limited to this runInstance method.